gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractRegionMap.java

```diff
@@ -1216,7 +1216,7 @@ public final boolean initialImagePut(final Object key,
                 VersionStamp stamp = null;
                 VersionTag lastDeltaVersionTag = null;
                 if (indexUpdater != null) {
```

Comment:
I guess we should modify this:

if (indexUpdater != null)
to
if (indexUpdater != null and indexsize > 0) ... [ Probably a size interface method would need to be added ]